src/expr.rs

```diff
@@ -1921,7 +1926,22 @@ fn rewrite_call_args(
     // Try letting the last argument overflow to the next line with block
     // indentation. If its first line fits on one line with the other arguments,
     // we format the function arguments horizontally.
-    let tactic = try_overflow_last_arg(&item_context, &mut item_vec, args, shape, one_line_width);
+    let tactic = if args.len() > 0 && args[0].is_expr() {
+        let args = args.iter().map(|e| e.to_expr()).collect::<Vec<_>>();
```

Comment:
Rather than use `is_expr` and an `if`, could you pass a closure from the client?